### DISCUSSION TITLE
[Question]: replace Orchard Storage Implementation

### DISCUSSION THREAD

**Discussion by JanEggers** [OrchardCMS/OrchardCore#15848] [Q&A]
hi there.

I am currently building an app based on ms orleans and need some configuration UI.

The idea is that I use the UI part of Orchard to configure the Grains inside the Orleans cluster.

So what I would like to try is create static Content Types that match the grain state i need to configure and then override OrchardCore.ContentManagement.Abstractions/IContentManager
to call into my orleans cluster instead of a db.

is IContentManager the correct thing to implement in such a case or is there a lower level abstraction that I should have a look at?

**Comment by jptissot** [comment]
@jpiquot probably has input on this question. See also https://github.com/OrchardCMS/OrchardCore/issues/2875
